src/adagrams.js

```diff
@@ -1,7 +1,113 @@
+let letterPool = ['A', 'A', 'A', 'A', 'A', 'A', 'A', 'A', 'A', 'B', 'B', 'C', 'C', 'D', 'D', 'D', 'D', 'E', 'E', 'E', 'E', 'E', 'E', 'E', 'E', 'E', 'E', 'E', 'E', 'F', 'F', 'G', 'G', 'G', 'H', 'H', 'I', 'I', 'I', 'I', 'I', 'I', 'I', 'I', 'I', 'J', 'K', 'L', 'L', 'L', 'L', 'M', 'M', 'N', 'N', 'N', 'N', 'N', 'N', 'O', 'O', 'O', 'O', 'O', 'O', 'O', 'O', 'P', 'P', 'Q', 'R', 'R', 'R', 'R', 'R', 'R', 'S', 'S', 'S', 'S', 'T', 'T', 'T', 'T', 'T', 'T', 'U', 'U', 'U', 'U', 'V', 'V', 'W', 'W', 'X', 'Y', 'Y', 'Z']
+
+function shuffle(a) {
+  let j, x, i;
+  for (i = a.length - 1; i > 0; i--) {
+      j = Math.floor(Math.random() * (i + 1));
+      x = a[i];
+      a[i] = a[j];
+      a[j] = x;
+  }
+  return a;
+}
+
 const Adagrams = {
   drawLetters() {
-    // Implement this method for wave 1
+    shuffle(letterPool);
+    return letterPool.slice(0, 10);
   },
+
+  usesAvailableLetters(input, lettersInHand) {
+    let lettersObject = {};
+
+    // Put lettersIn Hand into a hash-like object
+    lettersInHand.forEach((letter) => {
+      if (lettersObject[letter] === undefined) {
+        lettersObject[letter] = 1;
+      } else {
+        lettersObject[letter] += 1;
+      }
+    });
+
+    // Index through each letter and compare to hash-like object
+    let i = 0
+    while (i < input.length) {
+      if (lettersObject.hasOwnProperty(input[i]) && lettersObject[input[i]] > 0) {
+        lettersObject[input[i]] -= 1;
+        i += 1
+      } else {
+          return false;
+        }
+    }
+    return true;
+   },
+
+  scoreWord(word) {
+    word = word.toUpperCase();
+    const scoreTable = {
```

Comment:
Style note: move this object `scoreTable` to outside the function `scoreWord`, but inside the module